pkg/helm/factory.go

```diff
@@ -0,0 +1,125 @@
+package helm
+
+import (
+	"context"
+	"sync"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/controller/clusters"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// Config contains configuration options for Helm.
+type Config struct {
+	RegistryMirror *registrymirror.RegistryMirror
+	Env            map[string]string
+	Insecure       bool
+}
+
+// Opt is a functional option for configuring Helm behavior.
+type Opt func(*Config)
+
+// ExecuteableClient represents a Helm client.
+type ExecuteableClient interface {
+	PushChart(ctx context.Context, chart, registry string) error
+	PullChart(ctx context.Context, ociURI, version string) error
+	ListCharts(ctx context.Context, kubeconfigFilePath string) ([]string, error)
+	SaveChart(ctx context.Context, ociURI, version, folder string) error
+	Delete(ctx context.Context, kubeconfigFilePath, installName, namespace string) error
+	UpgradeChartWithValuesFile(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, valuesFilePath string, opts ...Opt) error
+	InstallChartWithValuesFile(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, valuesFilePath string) error
+	InstallChart(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, namespace, valueFilePath string, skipCRDs bool, values []string) error
+	Template(ctx context.Context, ociURI, version, namespace string, values interface{}, kubeVersion string) ([]byte, error)
+	RegistryLogin(ctx context.Context, registry, username, password string) error
+}
+
+// RegistryClient represents a Helm client.
+type RegistryClient interface {
```

Comment:
confused what this is for?

Reply:
I mainly created this interface to scope down the methods available to the client returned by the factory, as only `RegistryLogin` and `Template` is called on it at some point.

`ExecuteableClient` is more general interface that has more methods due to the other consumers of `BuildHelmExecutable` 

If we don't need it though, I can remove.

Reply:
Talked offline, we are going with one interface for now